### ISSUE TITLE
Atlas - option to create separate workspaces by users. Organization workspace vs user workspace

### ISSUE THREAD

**Issue by gowthamrao** [OHDSI/Atlas#724]
Not sure if this is possible - but putting some thoughts for a potential future effort.

Every authenticated user within an organization may have a separate work-space, like having an independent results schema. This is in addition to having a organization workspace that is shared across users within the organization.

User workspace are private to an individual user. Within a workspace user have ability to create cohort, cohort definition, concept-sets, incident rate etc, and either share or not share with other users as determined by the organization administrator. Only shared entities are visible to other users, and may be filtered. These are managed by RBAC.

Organization workspace are shared across users within the organization. This is similar to current Atlas. 

An Atlas instance may have many organizations. An individual user may belong to more than one organization at the same time.

Advantages: one user can work independently without overwriting the work of another user. Once finalized, the final work may be published to the global workspace. This process may be managed and governed.

**Comment by gklebanov** [comment]
Gowtham -  the ideas are very good, and needed indeed! I wanted to share some ideas, thoughts and work that we have been doing in this space as well.

When building IQVIA Research Network, we have been brainstorming how to make both ARACHNE and ATLAS multi-tenancy e.g. multiple organizations as well as improve support for an individual user do things in an isolated environment. What we have been able to do so far:

**ARACHNE**:
 - Added support for **multiple tenants**. Each tenant (a.k.a organization) will have a separate user base, data catalog, studies (collection of analysis done by a group of individuals), insights and individual workspaces
 - It already supports **studies**, e.g. multiple individuals collaborating on a single research via a series of analyses
 - finishing **individual workspaces** e.g. isolated environment where an individual can execute a series of analysis outside of the study

**ATLAS**
 - improved integration between ARACHNE and ATLAS
 - at this point, due to complexities of changes, we opted on using **multiple instances to support multiple organizations**. Each ATLAS instance can authenticate against a tenant in ARACHNE
- improved security to avoid users changing someone else's artifacts
- ATLAS users will soon be able to push and execute their individual analyses into ARACHNE Workspace, where they can execute it across data sets in one organization as well as across the network but in an isolated fashion

We also started brining up execution and results parts between ARACHNE and ATLAS into a single component/design - this is critical to get consistency across those two systems.

Some lessons from that exercise - ATLAS support for multi tenancy and individual workspaces proved to be a bit more difficult due to the significant refactoring it would require, we said that it will be an iterative journey overtime as we are continuously improving it. I also think that we need to start away from RESULTS schema as a permanent concept - this should just a temporary construct that would allow execution of analysis and storing of transient data but the results should be somewhere - e.g. results repository -  where it could be designed right from the start to better support isolation between runs, users, studies etc..
